site/source/_posts/2018/03/11/Dojo-2-rc1.md

```diff
@@ -0,0 +1,195 @@
+
+By default, Dojo's custom elements are built for evergreen browsers to reduce overall bundle size, meaning that most basic widgets are <20KB (gzipped) which includes all of the `@dojo` runtime needed to support the custom element.
+
+The `customElement` decorator annotates the widget class to convert it to a custom element:
+
+```typescript
+interface HelloWorldWidgetProperties {
+	onClick: (event: Event) => void;
+	foo: string;
+	bar: string;
+}
+
+@customElement<HelloWorldWidgetProperties>({
+	tag: 'hello-world',
+	attribute: [ 'foo', 'bar' ],
+	events: [ 'onClick' ]
+})
+class HelloWorldWidget extends WidgetBase<HelloWorldWidgetProperties> {
+// ...
+}
+```
+
+To compile a Dojo widget into a web component, run the following command:
+
+```
+npx @dojo/cli build widget --elements src/HelloWorld
+```
+
+For ease of use, all [`@dojo/widgets`](http://github.com/dojo/widgets) are compiled to custom elements and published to npm for easy import into your applications.
+
+Learn more about creating your first Dojo widgets in the [first Dojo 2 application tutorial](https://dojo.io/tutorials/001_static_content/).
+
+## The Dojo CLI
+
+CLI tools provide an easy way to work with frameworks, a trend that was popularized in the JS community by Ember. The [Dojo CLI](http://github.com/dojo/cli) is a collection of packages, providing a CLI architecture which enables a modular approach to create and work with commands.
+
+One of our goals in creating an easy out of the box experience with Dojo is to abstract users away from complex configuration by providing intelligent defaults internally. These defaults are easily overridden by experienced users who prefer something beyond our recommended default options.
+
+To install the Dojo CLI, run the following command:
+
+`npm install -g @dojo/cli`
+
+The [`@dojo/cli-build-app`](http://github.com/dojo/cli-build-app) package provides powerful tools to code split dynamically imported widgets automatically. Support is planned for the near future to enable users to specify this purely by configuration.
+
+`@dojo/cli-build-app` also offers basic support for Build Time Rendering. During an application's build step, the command renders the application, extracts the resulting HTML into the application's index.html, and inlines critical CSS. Build time rendering is an early minimal viable version with more refinements to come. Please let us know what feedback you may have!
```

Comment:
Shall we show how to configure build time rendering in the `.dojorc`?

---

To enable build time rendering simply add the following configuration to your projects `.dojorc`, include a root element with an `id` (referenced as the `root` in the config below) in the `index.html` and use `projector.merge` over `projector.append`:

```json
{
  "build-app": {
    "build-time-render": {
      "root": "app"
    }
  }
}
```